### ISSUE TITLE
Linux will not start after installing. Three other users have had this problem: "Fatal server error: (EE) AddScreen/ScreenInit failed for driver 1"

### ISSUE THREAD

**Issue by runner18** [dnschneid/crouton#3291]
```
Please paste the output of the following command here: 
EGL_MESA_drm_image required.
(EE)
Fatal server error:
(EE) AddScreen/ScreenInit failed for driver 1
(EE)
(EE)
Please consult the The X.Org Foundation support
at http://wiki.x.org
for help.
(EE) Please also check the log file at "/tmp/Xorg.crouton.1.log" for additional information.
(EE)
(EE) Server terminated with error (1). Closing log file.
```

#### Please describe your issue:
After installing, kde will not start.
Multiple users have had similar issues.
https://github.com/dnschneid/crouton/issues/3235
https://github.com/dnschneid/crouton/issues/3276
https://github.com/dnschneid/crouton/issues/3213

#### If known, describe the steps to reproduce the issue:
After successfully installing kde and setting up a username and password, I start kde, the screen goes black, and it returns to chrome os with an error message. This has happened for xfce as wellTh

Thank you!

**Comment by runner18** [comment]
Okay I've looked into it some more, and I've found some possible solutions that I have yet to try but have worked for others.

**Solution #1**

The reason it doesn't work is because the method 'xorg' can't get the screen to move over to Linux.

Fortunately, there's the 'xiwi' method, which lets you open Linux through a browsing tab or window. It's faster and will (hopefully) work better. So that's cool.

Make sure to get [this extension](https://chrome.google.com/webstore/detail/crouton-integration/gcpneefbbnfalgjniomfjknbcgkbijom)!

Change the downloading script to
`sudo sh ~/Downloads/crouton -t xfce,xorg,xiwi`

Then when you go to start it up, you have more options to open Linux that could work.

`sudo startxfce4` (that will launch in a separate screen via 'xorg', the default)

`sudo startxfce4 -X xiwi` (that will launch in a window via 'xiwi')

`sudo startxfce4 -X xiwi-fullscreen` (that will launch fullscreen via 'xiwi')

`sudo startxfce4 -X xiwi-tab` (that will launch in a browsing tab)

Thanks to https://github.com/DennisLfromGA

 **Solution #2**

 This is voodoo magic to me, but here you go.

> 
> OK.
> 
> Download this file to your "Downloads" directory in ChromeOS:
> [20-armsoc.txt](https://github.com/dnschneid/crouton/files/121172/20-armsoc.txt)
> 
> In crosh window, run shell command to launch bash-like shell
> 
> Rename downloaded file from .txt to conf:
> 
> ```
> cd ~/Downloads
> mv 20-armsoc{.txt,.conf}
> ```
> Login into chroot without X11:
> 
> sudo enter-chroot -n precise
> Move that file into proper place and flush the RAM cache:
> 
> ```
> sudo mv ~/Downloads/20-armsoc.conf /usr/share/X11/xorg.conf.d/
> sync
> ```
> Logout chroot with "Ctrl + d" or run exit command
> 
> Re-login into chroot with X11/freon:
> 
> `sudo startxfce4`
> That's all, maybe...
>
> Regards.

Thanks to
https://github.com/tista500

**Comment by OsiRex1020** [comment]
For method 1, do I have to funny uninstall the existing files and stuff? and if so, how do I do so?

**Comment by OsiRex1020** [comment]
I've done method 1. Now I have a new problem altogether. I'm getting this:
![screenshot 2017-06-22 at 7 25 18 am](https://user-images.githubusercontent.com/29465303/27420048-073f1638-571c-11e7-9d24-97879cf5fe32.png)

I've looked for a way to solve this, but didn't find any useful articles, except one using he bootstrap and apt-get; both of which didn't work.

**Comment by runner18** [comment]
Okay, so finally figured out how to get crouton to work. So when you download your system, don't even include "xorg" in your list of targets. It's evil. Just put extension and xiwi. (By targets I mean the things in the download scripts after "-t")

**Comment by runner18** [comment]
So like:
sudo sh ~/Downloads/crouton -t xfce, xiwi, extension

The extension has yet to fail me. :-)
Make the sure the extension it turned on btw.

**Comment by OsiRex1020** [comment]
Thanks. I got it done. 
Sorry I didn't reply sooner.

**Comment by runner18** [comment]
No prob glad it worked. 👍

**Comment by Fireflaker** [comment]
This solution does not seem to work for Samsung ARM-based Chromebooks. Further tutorials will help a lot. It is possible(I did it somehow) but not easy.

**Comment by specious** [comment]
I just installed crouton on a ARM Samsung Series 3 Chromebook and was just as puzzled as everyone else at xorg not starting, with `failed to load driver: exynos`.

I first tried to chroot and run `sudo apt-get install xserver-xorg-video-armsoc-exynos` but apt complained about unresolved dependencies and I went looking for a quicker solution.

Downloading and placing `20-armsoc.conf` into `/usr/share/X11/xorg.conf.d/` according to @runner18's instructions fully resolved this hurdle.
